app/helpers/conditions_helper.rb

```diff
@@ -135,105 +148,33 @@ def section_title(section)
                  escape: false)
   end
 
-  def previous_phase?(current_question, dropdown_phase)
-    current_question.phase.number > dropdown_phase.number
-  end
-
-  def not_previous_section?(current_question, dropdown_section)
-    later_section = current_question.section.number < dropdown_section.number
-    later_phase = current_question.phase.number < dropdown_section.phase.number
-    not_last_section_question = (current_question.section.number == dropdown_section.number &&
-                                 current_question.number != dropdown_section.questions.size)
-    return later_section || later_phase || not_last_section_question
-  end
 
-  def not_previous_question?(current_question, dropdown_question)
-    later_question = current_question.number < dropdown_question.number
-    later_section = current_question.section.number < dropdown_question.section.number
-    later_phase = current_question.phase.number < dropdown_question.phase.number
-    return later_question || later_section || later_phase
-  end
-
-  # given a conditions array, group conditions by number as a hash
-  def group_show_conditions(conditions)
-    conditions_grouping = {}
-    conditions.each do |condition|
-      conditions_grouping.merge!(condition.number => [condition]) {|_op, cond1, cond2|
-          cond1 + cond2 if cond1.is_a?(Array) && cond2.is_a?(Array)
-      }
-    end
-    conditions_grouping
-  end
-
-  # ensures conditions of type 'remove' come first. conditions of type Condition
-  def conditions_ordered(conditions)
-    grouped_conditions = group_show_conditions(conditions)
-    grouped_conditions.each do |option, conditions|
-      conditions.sort_by{|condition| condition.action_type.to_s.length}
-    end
-    grouped_conditions
-  end
+  # used when displaying a question while editing the template
+  # converts condition into text
+  def condition_to_text(conditions)
```

Comment:
Another suggestion to make things even more readable:
This helper is doing a lot for different contexts. I would suggest moving the controller specific bits to controllers/concerns/ and these view specific bits to a presenter. I think helpers should only be used for generic view stuff like formatting dates, etc.